messaging/kafka-avro-reactive-messaging/src/test/java/io/quarkus/ts/messaging/kafka/BaseKafkaAvroGroupIdIT.java

```diff
@@ -0,0 +1,94 @@
+package io.quarkus.ts.messaging.kafka;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.sse.SseEventSource;
+
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.restassured.http.ContentType;
+
+abstract class BaseKafkaAvroGroupIdIT {
+    private static final int TIMEOUT_SEC = 25;
+    private static final int EVENTS_AMOUNT = 5;
+
+    private String endpoint;
+    private Client client = ClientBuilder.newClient();
+    private List<String> receive = new CopyOnWriteArrayList<>();
+    private boolean completed;
+    private Random rand = new Random();
+
+    @Test
+    public void testAlertMonitorEventStream() throws InterruptedException {
+        GivenSomeStockPrices(getAppA(), EVENTS_AMOUNT);
+        AndApplicationEndpoint(getEndpoint(getAppA()) + "/stock-price/stream");
+        whenRequestSomeEvents(EVENTS_AMOUNT);
+        thenVerifyAllEventsArrived();
+        // Application B should have a different auto-generated group ID so,
```

Comment:
What would happen when the ID wouldn't be different?

Reply:
The Groups IDs are like tasks "to-do". You could have one task with several consumers or workers subscribed to it. In this case, all the workers will have the same group ID(are going to work on the same task), so each worker will handle different events from this task/groupID. 

But sometimes you want to do different things, different actions over the same events (over the same data), in this case, you will have different group IDs (one per task), and each worker of each task will receive the same event. 

In the first scenario, if you have 5 events and two workers (two Kafka consumers), one consumer will handle 3 events and the other 2.  Because both have the same groupID are working on the same task, the events are not duplicated[1].

In the second scenario...let say "save some data into Postgresql and mongoDB" then you will have two groupdIDs
GroupID = pushToPostgresql
GroupID = pushToMongoDB

By this way if you have one consumer per groupID, each one will process the same amount of events. 

So, What would happen when the ID wouldn't be different?
the second consumer will have the offset in the last position of the first consumer and will not have anything to do (in this case) because is already processed. 

All of this could be getting more complex if you add into the equation the concept of "partitions". 

In theory, one consumer could read from one or more partitions (but two consumers can't read from the same partition with the same groupID). That could be very interesting in terms of scalability. I mean...In Kafka, if you want to scale, you have to create more partition of your topic and in this way, you could add more consumers. 

Another interesting topic is that all the events that are inside a partition are going to be processed in order. And all the events with the same Key (record key) are going to ends up in the same partition (of this topic). So, if you want to process some event in order in Kafka you should be careful in order to choose your key. 

Now if you join these two paragraphs (From a conceptual point of view), you will ask yourself one question... Ok, In kafka in order to scale I have to do a repartition (increase the amount of partition of one topic), but if my scenario requires processing the events in order, this repartition will help me?... again you have to choose well you "partition"-key (record key)...but yes I also ask this question to my self.... 

[1]. the events are not going to be "duplicated"...in a physical way (as far as I know) kafka never ever duplicated an event, is an immutable log with pointers(with offsets), but in a conceptual point of view, we could have the feeling that if you have two consumers with different groupID the events are duplicated (but is just a feeling).  

[Know more](https://quarkus.io/version/main/guides/kafka)